File: Data Structures & Algorithms/rotate-array/submission-1.go
// [1,2,3,4,_,6,7,8]  //5
// [5,2,3,4,1,6,7,8]  //5


func rotate(nums []int, k int) {
	for i:=0; i<k;i++ {
		end := nums[len(nums)-1]
		// nums[1:(len(nums)-2)] = nums[0:(len(nums)-2)]
		for i:=len(nums)-2; i>=0; i-- {
			nums[i+1] = nums[i]
		}
		nums[0] = end
	}
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
# [1,2,4,6]
# 246  146 126 124
#  642  64  6      -> [1, 6, 64,642]
#         1   12  124  
class Solution:
	def productExceptSelf(self, nums: List[int]) -> List[int]:
		pre :list[int] = [1,]  # reverse later
		suf :list[int] = [1,]
		for i in range(len(nums)-1):
			suf.append(suf[-1]*nums[i])
		for i in range(len(nums)-1,0,-1):
			pre.append(pre[-1]*nums[i])
		
		pre.reverse()

		return [x*y for x, y in zip(pre,suf)]